projects/igniteui-angular/src/lib/grids/common/crud.service.ts

```diff
@@ -686,7 +686,12 @@ export class IgxGridCRUDService extends IgxRowAddCrudState {
         this.endEdit(true, event);
 
         if (parentRow != null && this.grid.expansionStates.get(parentRow.key)) {
-            this.grid.collapseRow(parentRow.key);
+            const key = parentRow.key;
+            this.grid.collapseRow(key);
+            // detect changes and get correct row by key
+            // in case collapsing changes row context.
+            this.grid.cdr.detectChanges();
```

Comment:
Why do we really need to detect any changes here? The `dataView` used in `getRowByKey` is updated after the `collapseRow` method. Is there a specific case that this is fixing. Also, what do you think of handling this with a copy of the parentRow since it comes from the action strip context and might be changed. Thus, something like this 
` const row = Object.assign(copyDescriptors(parentRow),parentRow);`
And then use this `row` instance further so that it doesnt change?

Reply:
I added the change detection it as precaution since nothing in the collapse logic does change detection nor pipe triggers and I wanted to be sure everything is up-to-date.

Also I don’t mind using a copy. I’m somewhat concerned with copying the whole IgxTreeGridRowComponent, since it’s a whole component object and the cloning of some of the angular specific props might fail or be incomplete. I’d much rather work with what the IgxTreeGridRow object that the API returns instead. But that’s just my preference.

Reply:
About the cloning the row object, we are cloning the `rowType` interface so I dont think there will be an issue but I undestand what are your concerns. I`m just not a big fan of invoking change detection in the middle of the method. 
Also, collapse logic does trigger change detection here: https://github.com/IgniteUI/igniteui-angular/blob/959435d7ab9b17720db1fd511349d5389fb92ef2/projects/igniteui-angular/src/lib/grids/grid-base.directive.ts#L4174.